integration-test/src/test/java/org/apache/iotdb/db/it/udf/IoTDBUDFSessionWindowQueryIT.java

```diff
@@ -0,0 +1,282 @@
+  }
+
+  private static void createTimeSeries() {
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      statement.execute("SET STORAGE GROUP TO root.vehicle");
+      statement.execute("CREATE TIMESERIES root.vehicle.d1.s3 with datatype=INT32,encoding=PLAIN");
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  private static void generateData() {
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      for (int i = 0; i < ITERATION_TIMES; ++i) {
+        if (i == 5 || i == 6 || i == 7 || i == 51 || i == 52 || i == 53 || i == 54 || i == 9996
+            || i == 9997 || i == 9998) {
+          continue;
+        }
+        statement.execute(
+            (String.format("insert into root.vehicle.d1(timestamp,s3) values(%d,%d)", i, i)));
+      }
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  private static void registerUDF() {
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      statement.execute(
+          "create function window_start_end as 'org.apache.iotdb.db.query.udf.example.WindowStartEnd'");
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  private void testSessionTimeWindowSS(
+      String sessionGap, long[] windowStart, long[] windowEnd, Long displayBegin, Long displayEnd) {
+    String sql;
+    if (displayBegin == null) {
+      sql =
+          String.format(
+              "select window_start_end(s3, '%s'='%s', '%s'='%s') from root.vehicle.d1",
+              UDFTestConstant.ACCESS_STRATEGY_KEY,
+              UDFTestConstant.ACCESS_STRATEGY_SESSION,
+              UDFTestConstant.SESSION_GAP_KEY,
+              sessionGap);
+    } else {
+      sql =
+          String.format(
+              "select window_start_end(s3, '%s'='%s', '%s'='%s', '%s'='%s', '%s'='%s') from root.vehicle.d1",
+              UDFTestConstant.ACCESS_STRATEGY_KEY,
+              UDFTestConstant.ACCESS_STRATEGY_SESSION,
+              UDFTestConstant.DISPLAY_WINDOW_BEGIN_KEY,
+              displayBegin.longValue(),
+              UDFTestConstant.DISPLAY_WINDOW_END_KEY,
+              displayEnd.longValue(),
+              UDFTestConstant.SESSION_GAP_KEY,
+              sessionGap);
+    }
+
+    try (Connection conn = EnvFactory.getEnv().getConnection();
+        Statement statement = conn.createStatement();
+        ResultSet resultSet = statement.executeQuery(sql)) {
+      assertEquals(2, resultSet.getMetaData().getColumnCount());
+      for (int i = 0; i < windowStart.length; i++) {
+        resultSet.next();
+        Assert.assertEquals(resultSet.getLong(1), windowStart[i]);
+        Assert.assertEquals(resultSet.getLong(2), windowEnd[i]);
+      }
```

Comment:
use while(resultSet.next()) and assert cnt instead.

Reply:
I've changed it.